Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractRandomCutTree.java

```diff
@@ -494,8 +497,8 @@ public PointReference addPoint(PointReference pointReference, long sequenceNumbe
             savedSiblingNode = followReference;
 
             // now iterative proceed up the tree and try to construct a cut
+            // currentUnmergedBox != null
```

Comment:
Did you mean to delete this?

Reply:
Same as above. This is a validation step.

Reply:
This looks like leftover code, because it doesn't invoke `checkState`, it doesn't throw an exception if the condition is false, etc.

Reply:
Oops - actually the next line which is removed had the checkArgument(). Let's make both of these asserts()